src/core/lib/transport/promise_endpoint.cc

```diff
@@ -46,10 +46,10 @@ PromiseEndpoint::PromiseEndpoint(
 }
 
 PromiseEndpoint::~PromiseEndpoint() {
-  // Last write result has not been polled.
-  GPR_ASSERT(!write_result_.has_value());
-  // Last read result has not been polled.
-  GPR_ASSERT(!read_result_.has_value());
+  // Promise endpoint close when last write result has not been polled.
+  if (write_result_.has_value()) write_result_.reset();
```

Comment:
You can just always call reset(). It will check internally if there is any value

Reply:
Done.